hnn_core/tests/test_network.py

```diff
@@ -329,29 +333,47 @@ def test_network():
         net.add_connection(**kwargs)
 
     # Test net.pick_connection()
-    kwargs_default = dict(net=net, src_gids=[0, 1], target_gids=[0, 1],
-                          loc=['soma', 'distal'], receptor=['ampa', 'gabaa'])
+    kwargs_default = dict(net=net, src_gids=None, target_gids=None,
+                          loc=None, receptor=None)
 
-    # List of tuples with (arg, item, correct_indices)
     kwargs_good = [
-        ('src_gids', 0, [30, 39]),
-        ('src_gids', 'L2_pyramidal', [29]),
-        ('src_gids', range(2), [30, 39]),
-        ('src_gids', None, [4, 29, 30, 39]),
-        ('target_gids', 35, [22, 34, 35, 37, 38, 40, 41, 42, 43, 44]),
-        ('target_gids', range(2), [30, 39]),
-        ('target_gids', 'L2_pyramidal',
-         [22, 34, 35, 37, 38, 40, 41, 42, 43, 44]),
-        ('target_gids', None,
-         [22, 28, 30, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44]),
-        ('loc', 'soma', [30, 39]),
-        ('loc', None, [30, 39]),
-        ('receptor', 'gabaa', [30, 39]),
-        ('receptor', None, [30, 39])]
-    for arg, item, indices in kwargs_good:
+        ('src_gids', 0),
+        ('src_gids', 'L2_pyramidal'),
+        ('src_gids', range(2)),
+        ('src_gids', None),
+        ('target_gids', 35),
+        ('target_gids', range(2)),
+        ('target_gids', 'L2_pyramidal'),
+        ('target_gids', None),
+        ('loc', 'soma'),
+        ('loc', None),
+        ('receptor', 'gabaa'),
+        ('receptor', None)]
+    for arg, item in kwargs_good:
         kwargs = kwargs_default.copy()
         kwargs[arg] = item
-        assert indices == pick_connection(**kwargs)
+        indices = pick_connection(**kwargs)
+        for conn_idx in indices:
+            if (arg == 'src_gids' or arg == 'target_gids') and \
+                    isinstance(item, str):
+                assert np.all(np.in1d(net.connectivity[conn_idx][arg],
+                              net.gid_ranges[item]))
+            elif item is None:
+                pass
+            else:
+                assert np.any(np.in1d([item], net.connectivity[conn_idx][arg]))
+
+    # Check that a given gid isn't present in any connection profile that
+    # pick_connection can't identify
+    conn_idxs = pick_connection(net, src_gids=0)
+    for conn_idx in range(len(net.connectivity)):
+        if conn_idx not in conn_idxs:
+            assert 0 not in net.connectivity[conn_idx]['src_gids']
+
+    # Check that pick_connection returns empty lists when searching for
+    # a drive targetting the wrong location
+    assert not pick_connection(net, src_gids='evdist1', loc='proximal')
```

Comment:
use `len` when checking for empty list. `not` is for boolean.

```suggestion
    conn_idxs = pick_connection(net, src_gids='evdist1', loc='proximal')
    assert len(conn_idxs) == 0
```